### PR TITLE
feat(aurelia-command): Add New Command API for aurelia-commands

### PR DESCRIPTION
**Pull request by joelcoxokc**
This PR replaces commanderjs with aurelia-command.

I have also modified the `aurelia.init()` and `aurelia.command()` from `src/index.js`

I have also placed `argv` on `process.AURELIA`

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -17,38 +16,32 @@ class Aurelia {
...
-    let newCmd = new NewCommand(program, this.config, this.logger);
-    let generateCmd = new GenerateCommand(program, this.config, this.logger);
-
-    this.commands[bundle.commandId] = bundle;
-    this.commands[init.commandId] = init;
-    this.commands[newCmd.commandId] = newCmd;
-    this.commands[generateCmd.commandId] = generateCmd;
-
+    program.init(config);
+    program.register(BundleCommand);
```
Comment (ahmedshuhel) [inline src/index.js]: @joelcoxokc Is it possible to have it like: `program.register(BundleCommand, this.config, this.logger)` This way we/framework will have the control what to inject in the command.

Comment (ahmedshuhel) [inline src/index.js]: Overall this looks good to me. :+1: 
@PWKad @zewa666 please let us know your views about this PR. We need another vote before we can start merging.

**Comment by zewa666** [comment]
looks good for me as well, happy that handlebars made it back into the package.json file. Ok from my side :+1:

File: src/index.js (hunk 1 above)
Comment (joelcoxokc) [inline src/index.js]: @PWKad I am not sure how to squash a couple commits. Let me know, because I hate having to re commit for tiny little changes.

Comment (joelcoxokc) [inline src/index.js]: @ahmedshuhel I believe it is possible to inject from the registration. I know that I am injecting the config and logger by default, But you are right. This needs to provided from the registration.

**Comment by joelcoxokc** [comment]
Merging PR. I will address @ahmedshuhel comment on injecting from registration in a new PR.

**Pull request merged**